### PR TITLE
Fixing the crash reporting and enable green tea test run for bare metal

### PR DESCRIPTION
**Pull request by rajkan01**
### Summary of changes 
Added missing mbed_error_initialize function call in bare metal boot code which is used
to retrieve the crash report from RAM in case of any previous abnormal re-boot; this function call is present in RTOS boot code.

### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [x] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 
@evedon @hugueskamba 

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: platform/source/mbed_sdk_boot.c [hunk 1]
```diff
@@ -76,6 +76,7 @@ int $Super$$main(void);
 int $Sub$$main(void)
 {
     mbed_main();
+    mbed_error_initialize();
```
Comment (evedon) [inline platform/source/mbed_sdk_boot.c:79]: How about mbed_toolchain_init()? Isn't that missing as well? I think we talked about this already - what was the outcome?

**Comment by hugueskamba** [comment]
How does the PR does `...enable green tea test run for bare metal`?

**Comment by hugueskamba** [comment]
Where can I find the RTOS boot code?

File: platform/source/mbed_sdk_boot.c (hunk 1 above)
Comment (rajkan01) [reply platform/source/mbed_sdk_boot.c:79]: We thought to reach martin to get clarification but still yet to reach martin.

**Comment by rajkan01** [comment]
> How does the PR do ...enable green tea test run for bare metal?

We guarded test case file with !defined(MBED_RTOS_CONF_PRESENT) and removed now to enable the test

**Comment by rajkan01** [comment]
> Where can I find the RTOS boot code?

https://github.com/ARMmbed/mbed-os/blob/master/rtos/source/TARGET_CORTEX/mbed_boot.c#L89

File: platform/source/mbed_sdk_boot.c (hunk 1 above)
Comment (hugueskamba) [reply platform/source/mbed_sdk_boot.c:79]: Have you reached him and he hasnt replied yet or have you not reached him yet?
If you are planning on reaching him, when is that likely to happen? I do not see immediately why it should not also be there.

Comment (rajkan01) [reply platform/source/mbed_sdk_boot.c:79]: I reached him today, still waiting for his response

**Comment by hugueskamba** [comment]
Is it possible to have a single function and put it in one place it can be executed with or without RTOS?

**Comment by rajkan01** [comment]
> Is it possible to have a single function and put it in one place it can be executed with or without RTOS?

I thought but it creates one more abstract layer and not sure which is accepted, my option we have to create common abstract on boot code kind of function pointer table and updating the function pointer at the compile-time based on boot(RTOS or Non-RTOS) with respective function.

File: platform/source/mbed_sdk_boot.c (hunk 1 above)
Comment (evedon) [reply platform/source/mbed_sdk_boot.c:79]: Russ was involved with most of these changes. Worth asking him on slack

Comment (evedon) [reply platform/source/mbed_sdk_boot.c:79]: Ticket created to investigate this point.

**Review by evedon — APPROVED** [review]

**Review by hugueskamba — APPROVED** [review]

**Pull request merged**